### PR TITLE
IGNITE-15450 Add special option to run snapshot commands synchronously.

### PR DESCRIPTION
**Pull request by xtern**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotSubcommand.java [hunk 1]
```diff
@@ -17,64 +17,71 @@
...
-     * @return Task class name to execute.
-     */
-    public String taskName() {
-        return taskName;
+    /** {@inheritDoc} */
+    @Override public void parseArguments(CommandArgIterator argIter) {
+        cmdArg = argIter.nextArg("Expected snapshot name.");
+
+        if (argIter.hasNextSubArg())
+            throw new IllegalArgumentException("Invalid argument: " + argIter.peekNextArg() + '.');
```
Comment (ololo3000) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotSubcommand.java:73]: May be it is more clear to change 'Invalid argument` -> `Unexpected argument`?

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotRestoreCommand.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    @Override public void parseArguments(CommandArgIterator argIter) {
+        String snpName = argIter.nextArg("Expected snapshot name.");
+        VisorSnapshotRestoreTaskAction restoreAction = parseAction(argIter);
+        Set<String> grpNames = null;
+        boolean sync = false;
+
+        while (argIter.hasNextSubArg()) {
+            String arg = argIter.nextArg(null);
+
+            if (restoreAction != VisorSnapshotRestoreTaskAction.START)
```
Comment (ololo3000) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotRestoreCommand.java:67]: I think it worth to move this check after all arguments are parsed and change exception message to something like:
`CANCEL or STATUS snapshot actions are not compatible with --sync or --groups arguments.

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotCreateCommand.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        String snpName = argIter.nextArg("Expected snapshot name.");
+        boolean sync = false;
+
+        if (argIter.hasNextSubArg()) {
+            String arg = argIter.nextArg(null);
+
+            SnapshotCreateCommandOption option = CommandArgUtils.of(arg, SnapshotCreateCommandOption.class);
+
+            if (option == null)
+                throw new IllegalArgumentException("Invalid argument: " + arg + ". ");
```
Comment (ololo3000) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotCreateCommand.java:52]: Let's mention in the exception message what options are available. As it is done for `SnapshotRestoreCommand`.

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotCreateCommand.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    protected SnapshotCreateCommand() {
+        super("create", VisorSnapshotCreateTask.class);
+    }
+
+    /** {@inheritDoc} */
+    @Override public void parseArguments(CommandArgIterator argIter) {
+        String snpName = argIter.nextArg("Expected snapshot name.");
+        boolean sync = false;
+
+        if (argIter.hasNextSubArg()) {
```
Comment (ololo3000) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotCreateCommand.java:46]: Should we here handle the situation when the user specifies unknown options after the --sync?
E.g. --sync --hello. May be it worth to change this `if` to `while`?

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotSubcommand.java (hunk 1 above)
Comment (xtern) [reply modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotSubcommand.java]: Done, thanks.

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotRestoreCommand.java (hunk 2 above)
Comment (xtern) [reply modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotRestoreCommand.java]: Fixed error message, I see no reason to move this condition below.

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotCreateCommand.java (hunk 3 above)
Comment (xtern) [reply modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotCreateCommand.java]: Done, thanks.

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotCreateCommand.java (hunk 4 above)
Comment (xtern) [reply modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/snapshot/SnapshotCreateCommand.java]: Replaced `if` to `while`. Thanks.

**Review by ololo3000 — APPROVED** [review]

**Review by ololo3000 — APPROVED** [review]

**Pull request merged**